ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,94 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
+            if num_divisions * sub_face_width + (num_divisions-1) * (dist_breakup - sub_face_width) > wall_length:
+                num_divisions = math.floor(wall_length/dist_breakup)
+
+            is_odd = lambda num: num % 2
+            start_pt_x = (num_divisions / 2) * div_dist if is_odd(num_divisions) == 0 else (math.floor(num_divisions / 2) + 0.5) * div_dist
+
+            btm_div_pts = [[start_pt_x, 0, sill_height_final]]
+            remainder = wall_length - (div_dist * num_divisions)
+            total_dist = remainder/2
+
+            while total_dist < wall_length:
+                total_dist += div_dist
+                next_pt = (wall_length / 2) - total_dist
+                btm_div_pts.append([next_pt, 0, sill_height_final])
+
+            sub_face_lines_start = [[btm_div_pts[i], btm_div_pts[i+1]] for i in range(num_divisions)]
+
+            line_cent_pt = [[line[1][0]+((line[0][0]-line[1][0])/2), 0, line[0][2]]
+                            for line in sub_face_lines_start]
+            dist_cent_line = div_dist if num_divisions != 1 else wall_length
+
+            sub_face_line_scale = sub_face_width / div_dist
+
+            sub_face_lines_start = [[[x[0][0] - ((x[0][0]-y[0])*(1-sub_face_line_scale)), 0, x[0][2]],[x[1][0] + ((y[0]-x[1][0])*(1-sub_face_line_scale)), 0, x[0][2]]]
+                                    for x, y in zip(sub_face_lines_start, line_cent_pt)]
+
+            final_glz_coords = [[line[1], line[0], [line[0][0], 0, line[0][2] + sub_face_height_final], [line[1][0], 0, line[1][2] + sub_face_height_final]]
+                                for line in sub_face_lines_start]
+            glz_area = sub_face_height_final * sub_face_width * len(sub_face_lines_start)
+        else:
+            if sub_face_width > wall_length:
+                sub_face_width = wall_length
+            sub_face_lines_start = [[wall_length/2, 0, sill_height_final],[-wall_length/2, 0, sill_height_final]]
+
+            dist_cent_line = wall_length
+            sub_face_line_scale = sub_face_width / wall_length
+            line_cent_pt = [0,0,0]
+            new_start_pt = [sub_face_lines_start[0][0] - ((sub_face_lines_start[0][0]-line_cent_pt[0])*(1-sub_face_line_scale)), 0, sub_face_lines_start[0][2]]
+            new_end_pt = [sub_face_lines_start[1][0] + ((line_cent_pt[0]-sub_face_lines_start[1][0])*(1-sub_face_line_scale)), 0, sub_face_lines_start[0][2]]
+            sub_face_lines_start = [new_start_pt, new_end_pt]
+
+            glz_area = sub_face_height_final * sub_face_width
+            final_glz_coords = [
+                sub_face_lines_start[1], sub_face_lines_start[0], [sub_face_lines_start[0][0], 0, sub_face_lines_start[0][2] + sub_face_height_final],
+                [sub_face_lines_start[1][0], 0, sub_face_lines_start[1][2] + sub_face_height_final]
+            ]
+        glazing_ratio = glz_area/(wall_length * rect_height)
+        return self.sub_faces_by_ratio(glazing_ratio)
```

Comment:
I don't understand. You go through all of this effort to calculate the actual geometry of the sub-faces and then you just throw it away to use another method?  I thought the purpose of this method was to return a list of `Face3D` objects representing sub-faces at a given width and height. Is this just a placeholder for now?